### DISCUSSION TITLE
Scenario autoprovisioning

### DISCUSSION THREAD

**Discussion by fturibak** [ludeknovy/jtl-reporter#188] [General]
Hi there!

Is there a possibility of autoprovisioning of scenario (in case scenario does not exists in JTL_REPORTER ).

This will be benefit in flexibility of the tool ( does not require extra administration ). In case you produce some amount of scenarios you need to remember every time to create one in jtl_reporter tool.

Thanks

**Comment by ludeknovy** [comment]
Hi, no - there is no such an option currently. 
What could be done is to have per project settings allowing to create a scenario if it doesn't exist on new test run push. 
Is that what you meant, right ?

**Reply by fturibak** [reply]
Exactly.

But my opinion is that this should not be an project option cause it's introducing complexity in JTL usage. My opinion is that this should be imbedded in the JTL engine and should be default behavior.

```
if  scenario_exists() {
  continue()
else 
  create_scenario()
  continue()
end if
```

This way user is not obligated to think if flag needs to be raised or not.

**Reply by ludeknovy** [reply]
I will implement it, but it definitely will be controlled by some settings.

**Reply by ludeknovy** [reply]
https://github.com/ludeknovy/jtl-reporter/issues/190

**Reply by ludeknovy** [reply]
https://github.com/ludeknovy/jtl-reporter/releases/tag/v4.4.7
